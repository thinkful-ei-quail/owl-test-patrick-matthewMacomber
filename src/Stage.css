.wrapper {
  background-color: #404853;
  min-width: 100%;
  min-height: 100%;
  margin: 0px;
  padding: 0px;
  text-align: center;
}

.stage {
  display: flex;
  width: 100%;
  height: 100%;
  justify-content: space-evenly;
}

.UserStage {
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.UserStage div span {
  margin-right: 10px;
}

.user-box {
  display: flex;
  flex-direction: column;
  width: 300px;
  align-items: center;
  background-color: #29333F;
  color: white;
  margin: 10px 0 10px 0;
}

.user-box div {
  margin-top: 10px;
  margin-bottom: 10px;
}

svg {
  stroke: red;
  stroke-width: 5px;
}

.UserStage img {
  width: 50%;
  height: 50%;
  object-fit: cover;
  border-radius: 50%;
  background-color: #1304B1;
}

.current-user {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 250px;
  border-style: solid;
  background-color: #29333F;
}

.current-user svg {
  margin-top: 10px;
}

.current-user img {
  width: 50%;
  height: 50%;
  object-fit: cover;
  border-radius: 50%;
  background-color: #1304B1;
}

.current-user-wrapper {
  display:inline-block;
}