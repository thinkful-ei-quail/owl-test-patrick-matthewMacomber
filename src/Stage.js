import React from 'react';


function isOnStage(par) {
  if (par.onStage) {
    return <section key={par.id} className="UserStage">
        <div>
          <div>{par.name}</div>
          <div><img src="volumeline.jpg" alt="Mic level bar"/></div>
          <div><span role="img" aria-label="Speaker Emoji">&#128266;</span><span role="img" aria-label="Magnifiying Glass">&#128269;</span></div>
        </div>
        <img src={par.avatar} alt="Avatar"></img>
      </section>
  }
  return '';
}

const user = {avatar: 'anAvatar.jpg'};

function Stage(props) {
  return <div className="Stage">
    <div>
    {props.participants.map(par => {
      return isOnStage(par);
    })}
    </div>
    <div>
      <img src="volumeline.jpg" alt="Mic level bar"/>
      <img src={user.avatar} alt="Avatar" />
    </div>
  </div>
}

Stage.defaultProps = {
  participants: []
}

export default Stage;