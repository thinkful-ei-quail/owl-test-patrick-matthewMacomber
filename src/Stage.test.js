import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from './Stage';
import participants from './Store';

describe('Stage testing', () => {
  //Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Stage participants={participants}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Smoke test
  it('renders without props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Stage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Stage participants={participants}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
});