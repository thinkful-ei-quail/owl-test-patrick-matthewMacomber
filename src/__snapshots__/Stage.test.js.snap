// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage testing renders the UI as expected 1`] = `
<div
  className="Stage"
>
  <div>
    <section
      className="UserStage"
    >
      <div>
        <div>
          Koren Templeton
        </div>
        <div>
          <img
            alt="Mic level bar"
            src="volumeline.jpg"
          />
        </div>
        <div>
          <span
            aria-label="Speaker Emoji"
            role="img"
          >
            🔊
          </span>
          <span
            aria-label="Magnifiying Glass"
            role="img"
          >
            🔍
          </span>
        </div>
      </div>
      <img
        alt="Avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </section>
    
    
    
    <section
      className="UserStage"
    >
      <div>
        <div>
          Ashla Attwool
        </div>
        <div>
          <img
            alt="Mic level bar"
            src="volumeline.jpg"
          />
        </div>
        <div>
          <span
            aria-label="Speaker Emoji"
            role="img"
          >
            🔊
          </span>
          <span
            aria-label="Magnifiying Glass"
            role="img"
          >
            🔍
          </span>
        </div>
      </div>
      <img
        alt="Avatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
    </section>
  </div>
  <div>
    <img
      alt="Mic level bar"
      src="volumeline.jpg"
    />
    <img
      alt="Avatar"
      src="anAvatar.jpg"
    />
  </div>
</div>
`;
